applications/system-service/wlan.h

```diff
@@ -50,6 +62,31 @@ class Wlan : public QObject, public SysObject {
             return 0;
         }
     }
+    signed int rssi(){
+        QDBusMessage message = m_interface->call("SignalPoll");
+        QMap<QString, QVariant> props;
+        const QVariant& arg = message.arguments().at(0).value<QDBusVariant>().variant();
+        const QDBusArgument& dArg = arg.value<QDBusArgument>();
+        dArg.beginMap();
+        while(!dArg.atEnd()){
+            QString key;
+            QVariant value;
+            dArg.beginMapEntry();
+            dArg >> key;
+            dArg >> value;
+            dArg.endMapEntry();
+            if(key == "rssi"){
+                props[key] = value;
+                break;
+            }
+        }
+        dArg.endMap();
```

Comment:
I suggest the following, which is a bit terser and does the same thing:
```suggestion
        auto props = qdbus_cast<QVariantMap>(message.arguments().at(0).value<QDBusVariant>().variant().value<QDBusArgument>());
```